exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/scan/v3/lifecycle/ReaderLifecycle.java

```diff
@@ -282,8 +302,8 @@ private void endBatch() {
     if (tableLoader.batchCount() == 1 || prevTableSchemaVersion < tableLoader.schemaVersion()) {
       reviseOutputProjection(tableLoader.outputSchema());
     }
-    buildOutputBatch(readerOutput);
-    scanLifecycle.tallyBatch();
+    int rowCount = buildOutputBatch(readerOutput);
```

Comment:
In other places, use `long` to define `rowCount`. So, this is simply the number of rows in a container, and it's not able to reach the `int` maximum, is that correct?

Reply:
In general, I tried to use `int` for the row count *within* a batch, and `long` for the row count *across* batches. Any one batch can have no more than 64K rows. But, a result set could potentially have 100B or 1T rows total across a large number of batches.

So, the limit for a scan (or for a reader) should be a `long`. Now, I'm not sure why anyone would use `LIMIT 4000000000`, but better safe than sorry.

If you see the use of `long` for batch counts, or `int` for result set counts, please call it out as it is probably an error.